app/lib/activitypub/tag_manager.rb

```diff
@@ -203,6 +203,39 @@ def uri_to_local_id(uri, param = :id)
     path_params[param]
   end
 
+  def uri_to_local_account_params(uri)
+    return unless local_uri?(uri)
+
+    path_params = Rails.application.routes.recognize_path(uri)
+
+    # TODO: handle numeric IDs
+    case path_params[:controller]
+    when 'accounts'
+      [:username, path_params[:username]]
+    when 'instance_actors'
+      [:id, -99]
+    end
+  end
```

Comment:
Unless I missed something it looks as if this method is only used inside this class here. So maybe it could be `private`?